### PR TITLE
CPRI-4191 Fetch Contributor Roles Based on AssetTypes

### PR DESCRIPTION
**Pull request by propattern**

### CODE REVIEW

File: src/services/taxonomy/__test__/getRoleTaxonomy.test.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import { expect } from 'chai'
 import sinon from 'sinon'
 // eslint-disable-next-line import/named
-import { getRoleTaxonomy, __RewireAPI__ as rewireAPI }
-  from '../getRoleTaxonomy'
+import { getRoleTaxonomy, __RewireAPI__ as rewireAPI } from '../getRoleTaxonomy'
 
-const expectedOutput = [
+const mdsRoels = [
```
Comment (sandeepnij90) [inline src/services/taxonomy/__test__/getRoleTaxonomy.test.js]: should this be roles?

File: src/utils/__test__/taxonomyUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  it('returns an empty object when it receives an empty array input', () => {
+    expect(remapTaxonomiesToInnerField([], '')).to.deep.equal({})
+  })
+
+  it('throws exception when the field is not blank', () => {
+    try {
+      remapTaxonomiesToInnerField(['a', 'b'], '')
+    } catch (e) {
+      expect(e.message).to.deep.equal('taxonomy cant be remapped to undefied or empty field')
```
Comment (Kerenhb) [inline src/utils/__test__/taxonomyUtils.test.js]: undefied _> undefined

File: src/utils/taxonomyUtils.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  if (isEmpty(mdsTaxonomies)) {
+    return {}
+  }
+
+  if (isEmpty(field)) {
+    throw Error('taxonomy cant be remapped to undefied or empty field')
+  }
+
+  const result = mdsTaxonomies.reduce((accOne, taxonomy) => {
+    const remapFields = taxonomy[field] || []
```
Comment (Kerenhb) [inline src/utils/taxonomyUtils.js]: Could have a better name here (remapFields), perhaps relating to the fact it's a key or an array?
Maybe taxonomyBasedKeys, as this is: ['https://schema.pearson.com/ns/assettype/text', 'https://schema.pearson.com/ns/assettype/richmedia', ...] correct?

File: src/utils/taxonomyUtils.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    const { id, value } = taxonomy
+
+    return remapFields.reduce((accTwo, remapFeild) => {
```
Comment (Kerenhb) [inline src/utils/taxonomyUtils.js]: remapFeild -> remapField (or if you follow the above comment something like taxonomyBasedKey)

File: src/utils/__test__/taxonomyUtils.test.js (hunk 2 above)
Comment (propattern) [reply src/utils/__test__/taxonomyUtils.test.js]: Done

File: src/utils/taxonomyUtils.js (hunk 4 above)
Comment (propattern) [reply src/utils/taxonomyUtils.js]: Done.

File: src/utils/taxonomyUtils.js (hunk 3 above)
Comment (propattern) [reply src/utils/taxonomyUtils.js]: Done.

File: src/utils/taxonomyUtils.js (hunk 4 above)
Comment (Kerenhb) [reply src/utils/taxonomyUtils.js]: I meant:
- remapFields -> taxonomyBasedKeys
- remapFeild -> taxonomyBasedKey

You seem to have done remapFields -> taxonomyBasedKey (should be plural still) and left remapFeild as is.

Comment (propattern) [reply src/utils/taxonomyUtils.js]: Pushed a fix

**Comment by Kerenhb** [comment]
@devu would this break any of your code (going from an array to a object)?

**Comment by devu** [comment]
no, multidimensional taxonomies are objects, approving and merging

**Pull request merged**